### PR TITLE
feat: Convert Metamask controller actions tests from mocha to jest

### LINKED ISSUE #19355: Finishing mocha tests transition
- [x] Account Import Strategies - 6 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/account-import-strategies/account-import-strategies.test.js
- [x] BackupController - 6 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/backup.test.js
- [x] CachedBalancesController - 4 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/backup.test.js
- [x] DetectTokensController - 9 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/cached-balances.test.js
- [x] EnsController - 7 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/cached-balances.test.js
- [ ] IncomingTransactionsController - 25 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/cached-balances.test.js
- [ ] MetaMetricsController - 34 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/cached-balances.test.js
- [x] preferences controller - 27 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/preferences.test.js
	- **Additional Step**: use Upper Camel case like other controller test suites
- [ ] SwapsController - 39 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/swaps.test.js
- [x] TransactionController - 84 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/swaps.test.js
- [x] Transaction state history helper - 6 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/transactions/lib/tx-state-history-helpers.test.js
- [x] txUtils (first iteration) - 27 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/transactions/lib/util.test.js
	- **Additional Step**: use more descriptive test suite name
- [x] PendingTransactionTracker - 19 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/transactions/pending-tx-tracker.test.js
- [x] txUtils (second iteration) - 4 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/transactions/tx-gas-utils.test.js
	- **Additional Step**: use more descriptive test suite name
- [x] TransactionStateManager - 37 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/controllers/transactions/tx-state-manager.test.js
- [x] multiple instances running detector - 5 tests 
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/detect-multiple-instances.test.js
- [ ] MetaMaskController (first iteration) - 18 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.actions.test.js
	- **Additional Step**: Use a more descriptive name
- [ ] MetaMaskController (second iteration) - 64 tests
	- **Location**: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/metamask-controller.test.js
	- **Additional Step**: Use a more descriptive name (dedupe)

### PR DESCRIPTION
**Pull request by itsyoboieltr**
## **Description**

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/25336?quickstart=1)

This PR converts `app/scripts/metamask-controller.actions.test.js` from mocha to jest. In particular, replaces instances of `assert` with `expect`, and removes `sinon` and uses `jest`'s mocking capabilities in its place. The `jest.config.js` was updated to include `app/scripts/metamask-controller.actions.test.js` in the `testMatch` array. The `.mocharc.js` was updated to ignore the test.

## **Related issues**

Fixes: #19355

## **Manual testing steps**

1. Running `yarn jest -- app/scripts/metamask-controller.actions.test.js` and seeing all tests pass
2. Running `yarn test:unit:mocha` and not seeing the metamask controller actions tests run.

## **Screenshots/Recordings**

Not applicable

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by itsyoboieltr** [comment]
Not ready to review / merge as some tests are failing with the following error message:

```
MetaMaskController › #addNewAccount › two parallel calls with same accountCount give same result

    pointHex must be hex string or Uint8Array

      3901 |     const releaseLock = await this.createVaultMutex.acquire();
      3902 |     try {
    > 3903 |       return await this.keyringController.createNewVaultAndKeychain(password);
           |              ^
      3904 |     } finally {
      3905 |       releaseLock();
      3906 |     }

```

Not sure what exactly is going wrong, as the logic of the test was not changed.

File: app/scripts/metamask-controller.actions.test.js [hunk 1]
```diff
@@ -138,21 +133,21 @@ describe('MetaMaskController', function () {
...
-      assert.equal(addNewAccountResult1, addNewAccountResult2);
+      expect(addNewAccountResult1).toStrictEqual(addNewAccountResult2);
     });
 
     it('two successive calls with different accountCount give different results', async function () {
       await metamaskController.createNewVaultAndKeychain('test@123');
       const addNewAccountResult1 = await metamaskController.addNewAccount(1);
       const addNewAccountResult2 = await metamaskController.addNewAccount(2);
-      assert.notEqual(addNewAccountResult1, addNewAccountResult2);
+      expect(addNewAccountResult1).not.toStrictEqual(addNewAccountResult2);
```
Comment (DDDDDanica) [inline app/scripts/metamask-controller.actions.test.js:150]: Can we try 
```suggestion
      expect(addNewAccountResult1).not.toEqual(addNewAccountResult2);
```
Should be sufficient

Comment (itsyoboieltr) [reply app/scripts/metamask-controller.actions.test.js:150]: The test fails anyway :(

```
  ● MetaMaskController › #addNewAccount › two successive calls with different accountCount give different results

    pointHex must be hex string or Uint8Array

      3914 |     const releaseLock = await this.createVaultMutex.acquire();
      3915 |     try {
    > 3916 |       return await this.keyringController.createNewVaultAndKeychain(password);
           |              ^
      3917 |     } finally {
      3918 |       releaseLock();
      3919 |     }

      at ensureBytes (node_modules/ethereum-cryptography/node_modules/@noble/curves/src/abstract/utils.ts:126:11)
      at Function.fromHex (node_modules/ethereum-cryptography/node_modules/@noble/curves/src/abstract/weierstrass.ts:318:55)
      at pubToAddress (node_modules/@ethereumjs/util/src/account.ts:285:52)
      at HdKeyring._addressfromPublicKey (node_modules/@metamask/eth-hd-keyring/index.js:306:7)
      at node_modules/@metamask/eth-hd-keyring/index.js:137:19
          at Array.map (<anonymous>)
      at HdKeyring.addAccounts (node_modules/@metamask/eth-hd-keyring/index.js:136:35)
      at KeyringController.addNewKeyring (node_modules/@metamask/keyring-controller/src/KeyringController.ts:738:21)
      at KeyringController.createKeyringWithFirstAccount_fn (node_modules/@metamask/keyring-controller/src/KeyringController.ts:1853:12)
      at KeyringController.createNewVaultWithKeyring_fn (node_modules/@metamask/keyring-controller/src/KeyringController.ts:1741:5)
      at KeyringController.createNewVaultAndKeychain (node_modules/@metamask/keyring-controller/src/KeyringController.ts:702:9)
      at MetamaskController.createNewVaultAndKeychain (app/scripts/metamask-controller.js:3916:14)
      at Object.<anonymous> (app/scripts/metamask-controller.actions.test.js:145:7)
```

Comment (itsyoboieltr) [reply app/scripts/metamask-controller.actions.test.js:162]: Fixed by enabling node environment using:

```typescript
/**
 * @jest-environment node
 */
```

**Review by danjm — APPROVED** [review]
LGTM! Great work

**Review by vthomas13 — APPROVED** [review]

**Pull request merged**